CRM/Core/Form.php

```diff
@@ -1215,7 +1215,7 @@ public function &addRadio($name, $title, $values, $attributes = [], $separator =
       // We use a class here to avoid html5 issues with collapsed cutsomfield sets.
       $optAttributes['class'] = $optAttributes['class'] ?? '';
       if ($required) {
-        $optAttributes['class'] .= ' required';
+        $optAttributes['required'] = TRUE;
```

Comment:
There was a deliberate reason for doing this due to some html5 issues iirc @demeritcowboy

Reply:
I think this change is the crux of the issue.  Without it, jQuery validation is not actually validating the field.  See [dev/core#2051](https://lab.civicrm.org/dev/core/-/issues/2051) for further explanation.  There is an issue with collapsed custom field sets where jQuery validation can't pry the accordion open to point out the required field.  It looks like nothing has happened, but the validation has failed.  However, this is a general problem with jQuery validation, not a problem with marking fields as required.

Reply:
I haven't run this just linking to what @seamuslee001 is referring to: https://github.com/civicrm/civicrm-core/pull/18080